includes/publisher/js/src/components/Email.jsx

```diff
@@ -16,6 +17,7 @@ export default function Email({
 		defaultValue: field.value,
 		required: field.required,
 		onChange: (event) => validate(event, field),
+		pattern: buildWildcardRegex(field.allowedDomains),
```

Comment:
Worth setting a fallback here in case `field.allowedDomains` is `undefined` for some reason?

```suggestion
		pattern: buildWildcardRegex(field.allowedDomains ?? ""),
```

I still had some email fields from before you added the `allowedDomains` property, which causes an empty form to display due to `values.split` failing in `buildWildcardRegex` (see below). A fallback might protect us from similar issues if `allowedDomains` isn't saved for some other reason. But feel free to ignore if you think it's overly defensive.

<img width="1553" alt="Screenshot 2022-04-15 at 09 44 22" src="https://user-images.githubusercontent.com/647669/163541227-678b2398-0555-46d6-b46f-38a2b703e8a1.png">

Reply:
I like this, in fact I had some defensive checks with the `buildWildcardRegex(...)` but removed unfortunately. May I make a suggestion though, instead of setting the default pattern  value to be `""` from the component(s), could I set default value within the `buildWildcardRegex(...)` function? 

Something like this :

```JavaScript
// ... brevity
export const buildWildcardRegex = (values = "", delimiter = ",") => {
// ...
```

Reply:
Sounds good, nice idea!